plugins/modules/ini_file.py

```diff
@@ -430,6 +490,9 @@ def do_ini(module, filename, section=None, option=None, values=None,
     if not within_section and state == 'present':
         ini_lines.append(u'[%s]\n' % section)
         msg = 'section and option added'
+        if 'section_has' in module.params:
```

Comment:
This check is always true, since `module.params['section_has']` is `None` when the option has not been specified.

You want to check for `module.params['section_has'] is not None`.

Reply:
Yes!